Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/WalletConnectKeyFlow.swift

```diff
@@ -0,0 +1,99 @@
+    }
+
+    var parameters: AddWalletConnectKeyParameters? {
+        keyParameters as? AddWalletConnectKeyParameters
+    }
+
+    init(completion: @escaping (Bool) -> Void) {
+        super.init(factory: WalletConnectKeyFlowFactory(), completion: completion)
+    }
+
+    override func didIntro() {
+        selectWallet()
+    }
+
+    func selectWallet() {
+        let vc = flowFactory.selectWallet { [unowned self] connection, wallet in
+            guard let address = connection.accounts.first else {
+                App.shared.snackbar.show(error: GSError.WCConnectedKeyMissingAddress())
+                return
+            }
+
+            keyParameters = AddWalletConnectKeyParameters(address: address,
+                                                          name: nil,
+                                                          connection: connection,
+                                                          wallet: wallet)
+            didGetKey()
+        }
+
+        show(vc)
+    }
+
+    override func doImport() -> Bool {
+        assert(parameters?.connection != nil)
+        assert(parameters?.wallet != nil)
+        assert(parameters?.name != nil)
+        guard OwnerKeyController.importKey(connection: parameters!.connection,
+                                           wallet: parameters!.wallet,
+                                           name: parameters!.name!) else {
+            WebConnectionController.shared.userDidDisconnect(parameters!.connection)
+            return false
+        }
+
+        return true
+    }
+}
+
+class WalletConnectKeyFlowFactory: AddKeyFlowFactory {
+    override func intro(completion: @escaping () -> Void) -> AddKeyOnboardingViewController {
+        let introVC = super.intro(completion: completion)
+        introVC.cards =  [
+            .init(image: UIImage(named: "ico-onboarding-import-key-1"),
+                    title: "How does it work?",
+                    body: "You can connect an owner key from another wallet. You will be asked to select it from a list of already installed wallets on your phone or you can display a QR code and scan it with another wallet."),
+
+            .init(image: UIImage(named: "ico-onboarding-import-key-2"),
+                    title: "How secure is that?",
+                    body: "WalletConnect is a secure protocol for exchanging messages. Safe app will not get access to your private key stored in your wallet."),
+
+            .init(image: UIImage(named: "ico-onboarding-import-key-3"),
+                    title: "Is my wallet supported?",
+                    body: "You wallet needs to support the WalletConnect protocol.")
+        ]
+        introVC.viewTrackingEvent = .connectOwnerOnboarding
+        introVC.navigationItem.title = "Connect Owner Key"
+        return introVC
+    }
+
+    func selectWallet(completion: @escaping (_ connection: WebConnection, _ wallet: WCAppRegistryEntry?) -> Void) -> SelectWalletViewController {
```

Comment:
let's switch the argument order in completion closure, and then we can pass the closure directly to the view controller.

Reply:
actually, we can remove this method, as it doesn't do anything useful